src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -323,6 +323,12 @@ function VisEditor($rootScope, $scope, $route, timefilter, AppState, $window, kb
       if (id) {
         notify.info('Saved Visualization "' + savedVis.title + '"');
         if ($scope.isAddToDashMode()) {
+          // We need to manually set the last url for this app so navigating back to visualize will open the saved
+          // visualization, not the unsaved one.
+          const savedVisualizationUrl =
+            kbnUrl.eval(`${kbnBaseUrl}#${VisualizeConstants.EDIT_PATH}/{{id}}`, { id: savedVis.id });
+          chrome.trackSubUrlForApp('kibana:visualize', savedVisualizationUrl);
```

Comment:
So, it *mostly* works, if I do this:
```
          const savedVisualizationUrl =
            kbnUrl.eval(`${VisualizeConstants.EDIT_PATH}/{{id}}`, { id: savedVis.id });

          kbnUrl.redirect(savedVisualizationUrl);
          // might need to use $rootScope here, $scope might be torn down before '$routeChangeSuccess' event
          const unlisten = $rootScope.$on('$routeChangeSuccess', () => {
            $window.history.pushState({}, '', `${chrome.getBasePath()}${kbnBaseUrl}#${savedVisualizationUrl}`);
            unlisten(); // we only want the first event
            const dashboardBaseUrl = chrome.getNavLinkById('kibana:dashboard');
            const dashUrlPieces = dashboardBaseUrl.lastSubUrl.match(/(.*)kibana#(.*)/);
            const dashSubUrl = `${dashUrlPieces[2]}&${DashboardConstants.NEW_VISUALIZATION_ID_PARAM}={{id}}`;
            kbnUrl.change(dashSubUrl, { id: savedVis.id });
          });
```

I still need to manually call `$window.history.pushState` to insert the extra url for some reason - probably because of the intercept and redirect, that url gets totally removed from history.  Without the pushState line, the backButton actually skips the unsaved url and goes back to the `/visualize/new/configure` url.  If I manually push the edit url into history, the back button will go back to the edit url, but an additional back button will take them back to the `new/configure` url instead of the unsaved version.  Not a huge deal, but technically incorrect because it skips the unsaved url.

Why do you prefer this version over the manual call to a new `trackSubUrlForApp` function?  I don't love the original solution, but this variation seems a bit hacky too, more complicated, and not exactly correct (though I suppose we could address that by manually pushing the unsaved url too...).